### ISSUE TITLE
When the roll back have finished, the primary display is not correctly applied.

### ISSUE THREAD

**Issue by sk666** [terrymacdonald/DisplayMagician#74]
**Describe the bug**
Hi Terry,

I have 2 problem since the latest version 2.1.5.5 when the roll back to the previous display profile occur:

1- when the roll back have finished, the primary display is not correctly applied. If my primary display is set to screen #1, after the roll back the primary display is now at the screen #2 even if windows say that the current primary display is #1 in the windows display settings menu. I don't know if you understand with my crappy english language, my task bar in w10 is set to have my quick launch bar and my notification area in my default primary display (screen #1). If I start display magician to start some programs with a game and switch to an another display profil and then roll back, my quick launch bar and my notification area have moved to the screen 2 even if windows show my that my screen #1 is actually my default primary display after the roll back. 

To fix this, I need to change my default primary display to screen 2 and after that, switch back to screen #1.

2- My second issue is something that you fixed for me in a previous a couple of months ago but problem is back with the latest version: 4 apps are supposed to launch when I start a game, but when I quit the game, the 4 programs are not fully closed as they are supossed to be. Sometimes 1 app still running after the roll back, sometimes 2 programs are running. I don't know if you can fix it again, I don't know if DM always check in the active process list if all of the 4 apps are still running during the roll back to force them to "shutdown".

**To Reproduce**
Steps to reproduce the behaviour:
1. Launch my customized profile of DM and after the game has started, quit the game. The 2 problems will occur after the roll back to my previous display profile.

**Expected behaviour**
1- if my default primary display is set to screen #1 before I use DM, it should be back to screen #1 after the roll back.

2- I have set DM to shutdown all the 4 apps after the roll back and DM miss 1 or 2 of them.

**Screenshots**
tell me if you need some screenshot of something.

**Reporting Logs**
If DisplayMagician starts:
- Open DisplayMagician and perform whatever actions are needed to make the bug occur.
- Click on the Settings button on the main DisplayMagician window
- Click on 'Create a Support Zip File' button
- Save the Support Zip File on your computer.
- Come back here, and upload the Support Zip File through your web browser and attach it to this issue.

If DisplayMagician does not:
- Open a terminal window and run `C:\Program Files\DisplayMagician\DisplayMagician.exe --trace` to create a TRACE-level DisplayMagician.log file.
- The above command will create a DisplayMagician.log file in `C:\Users\<yourusername>\AppData\Local\DisplayMagician\Logs` which should contain a LOT of detailed TRACE level log entries. 
- Come back here, and upload the DisplayMagician.log file through your web browser and attach it to this issue.
- Also attach the following files to this issue:
  - `C:\Users\<yourusername>\AppData\Local\DisplayMagician\Profiles\DisplayProfiles_2.1.json` 
  - `C:\Users\<yourusername>\AppData\Local\DisplayMagician\Shortcuts\Shortcuts_2.0.json`
  - `C:\Users\<yourusername>\AppData\Local\DisplayMagician\Settings_2.0.json`

**Enviroment (please complete the following information):**
 - Windows Version: [Win10]
 - DisplayMagician Version [2.1.5.5]
 - Video Card: [e.g. Asus NVIDIA GTX970]
 - Number and make/model of monitors: [2xDell 2208WFP]
 - Date of last video driver update: [v.497.29   2021-12-20]
 - Date of last windows update: [2022-01-02 (today)]

**Additional context**
[DisplayMagician-Support-20220102-1038.zip](https://github.com/terrymacdonald/DisplayMagician/files/7799395/DisplayMagician-Support-20220102-1038.zip)

I'm sorry for my bad english, please tell me if you don't undersatnd. Thank you very much for your help and for your amazing work with DM.

**Comment by terrymacdonald** [comment]
Hi @sk666 and welcome back!

I'm going to split these two issues up so I can track them separately. I'll paste your second issue below so I can refer to it to make the new issue:

2- My second issue is something that you fixed for me in a previous a couple of months ago but problem is back with the latest version: 4 apps are supposed to launch when I start a game, but when I quit the game, the 4 programs are not fully closed as they are supossed to be. Sometimes 1 app still running after the roll back, sometimes 2 programs are running. I don't know if you can fix it again, I don't know if DM always check in the active process list if all of the 4 apps are still running during the roll back to force them to "shutdown".

Thanks
Terry

**Comment by terrymacdonald** [comment]
@sk666 please go and follow issue #75 as well, so that you're notified when I post a comment in there for you to try. I'll look at this issue first a little later today.

**Comment by terrymacdonald** [comment]
@sk666 I've looked at your shortcut, and there are a few things setup incorrectly in it.

-  You do not need to start steam.exe in your start programs. In fact, that will cause an issue rather than fix things. When you select a game from the game launcher list, DisplayMagician knows how best to start that game and will automatically launch it the best way it knows how. In other words, DisplayMagician will start steam for you to launch American Truck Simulator.

-  You do not need to start DisplayMagican.exe in your start programs. That is probably the main cause of the broken system. **Never add DisplayMagician.exe to the list of start programs.** If you want to start DisplayMagician when the computer starts, then simply check the 'Start DisplayMagician minimised in notification area when the computer starts' checkbox on the main DisplayMagician window: 
![Screenshot 2022-01-04 133621](https://user-images.githubusercontent.com/5721772/147994847-63b43806-73ba-4820-b01a-3c60ccaa2bb5.png)

-  If you're still having issues after making the changes I've mentioned above, then American Truck Simulator may use a game launcher to start. Game launchers are little programs that start up do some checks, and then launch the main game executable. DisplayMagician tries to detect game launchers and try to monitor the main game executable, but sometimes it is unable to detect the game launcher, and it doesn't sit there and monitor the game. Instead, it returns straight away, before the game has been exited. 

If DisplayMagician thinks the game has been closed when it has not been closed (it waits for 30 seconds then stops monitoring), then you will need to tell it to monitor a different executable so it knows what executable to keep scanning for. I've tried to find out which common games use a game launcher, and I've listed them here: https://github.com/terrymacdonald/DisplayMagician/wiki/Main-Game-and-Application-Examples. I just checked ETS myself and it does not use a game launcher, so ATS should be fine. But if you find out later that ATS does need to monitor a different executable, then feel free to tell me so I can update the wiki page.

Hope that helps!
Thanks
Terry

**Comment by sk666** [comment]
Hi terry, 

I removed Steam and DM from the list of strated program and issue #74 AND #75 are still present. Steams and Simdashboard are still open after the game was closed and my default diplay have switched to screen #2 after the roll back. So both issue need to be reopened. All was working well before update to v2.1.5, should I need to downgrade DM?

Thank you

**Comment by terrymacdonald** [comment]
Hi,

Steam shouldn't be run from a  DisplayMagician start program.  It will
break DisplayMagician. If that worked in the past it was not intentional.
If you need Steam to be run on computer startup then I'd recommend doing
that through windows rather than DisplayMagician. DisplayMagician is
designed to run a game and start up any game related executables you want
and shut them down afterwards.

Can you please attempt to run your ATS shortcut, then (without closing
DisplayMagician) go to the settings screen and create a new Support Zip
file please? Then please upload it here again.

Hopefully the logs will show me what is not working.

Thanks
Terry

**Comment by terrymacdonald** [comment]
Reopening until further investigations continue.

**Comment by terrymacdonald** [comment]
This is a strange one @sk666. I will probably need to get you to perform some more testing with NVIDIAInfo (a standalone program I wrote which tests only the NVIDIA part of DisplayMagician) to see if there is something wrong, based on the description you gave.

Just to make sure I'm understanding you correctly:
- You start on a display profile A. Your primary display is display #1.
- You swap to display profile B. Your primary display is display #2.
- After you have swapped back to display profile A, your primary display is STILL display #2. (It should be display #1)

Is that correct?

If it is correct, can you please test that same display profile swap using the DisplayProfile page rather than via a Game Shortcut? I want to know if that makes it work or not.

Also, what is the name of display profile A (the first one) and display profile B (the second one) that you are using? I will be able to check the display profile config if I know the profile names, 

Do either display profile A or display profile B use NVIDIA surround?

Terry

**Comment by sk666** [comment]
Hi Terry,

sorry for the late reply, we have a couple time zone between us haha. Maybe this will confuse you a little more but I discovered something new after I deactivated Steam in my profile of DM. If I don't start steam in DM like I did before, I have a very weird bug with my mouse when I start the game with the desktop shortcut of my DM profile: if a program is open in one of my screen (ex: firefox), my mouse is locked in a part of the screen during game start and I need to open the task manager and do a rapid movement with the mouse and I need to click somewhere during this to unlock the mouse. It will be nice if I can have a video of this.

I don't know why DM is more stable if I start Steam on my profile but I think I should keep it because it's the only thing I found that resolve this bug. To prove my theory, I reactivated Steam on the tab 'choose waht happen before" and the bug is not present anymore after that. I don't understand this behavior after all that you said yesterday, it's very strange.

Anyway, to respond to your question, Profile B is nvidia surround (see attached screenshot for the names). You understand correctly, but I need to correct you in just one thing: After swap to my other profile display when I start the game (this second profile use nvidia surround), my primary display is still on my screen #1. I know this because if I start the game and switch back to my desktop when the game is still open, I see my notification area is still on screen #1, but I see the change during the roll back after I quit the game. 

I tried what you said, I swaped my display profile from the displayprofile page and the problem is not present when I did this but  the task bar disappear from screen #1 when I am on the surrund display (the task bar is only visible on screen #2 in this case).

After that, I tried to launch again the game with the DM sortcut and the behavior has changed. Now, if I use the shortcut, I see that DM have switched to my nvidia surround profile because I see the single complete taskbar in all the screen but the game appear only on my screen #1 and my screen #2 is on power save mode. The primary display is back to screen #1 after I quit the game (what I wanted) but the game don't use the 2 screen even if surround is activated. 

I tried to relaunch the game with the DM shortcut and now, everythings works fine. The game start with the surround, and when I quit the game, my primary display is back to screen #1 (what I wanted). The problem seem to be not present after I swaped my dispaly profile from the displayprofile page.

After that, I did an utlimate test: I rebooted my computer, and use again the DM shortcut to start the game and the game have not started with the dual screen, only with the screen #1 even if I am on the nvidia surround profile (and screen #2 fall in power save mode again). The good thing is my screen #1 is still the primary display but the bad thing is (for the first time), the task bar on screen #1 have moved at the top of screen #1. If I look in the windows settings to the task bar, it siwtched to "custom" instead of "bottom" (see attached screenshot). So I suspect it's a bug with nvidia surround itself, what do you think?

Do you still need a second support zip file?

Thank you for your help.

![display profile](https://user-images.githubusercontent.com/29050341/148073311-6d9d7ee4-d53a-4dcc-89ce-8b7f216397e2.PNG)
![task bar settings after the rool back](https://user-images.githubusercontent.com/29050341/148073313-eb633ac3-c89b-4f47-9fbb-c5c8cffdf221.PNG)

**Comment by terrymacdonald** [comment]
Hi Terry,

_"sorry for the late reply, we have a couple time zone between us haha. Maybe this will confuse you a little more but I discovered something new after I deactivated Steam in my profile of DM. If I don't start steam in DM like I did before, I have a very weird bug with my mouse when I start the game with the desktop shortcut of my DM profile: if a program is open in one of my screen (ex: firefox), my mouse is locked in a part of the screen during game start and I need to open the task manager and do a rapid movement with the mouse and I need to click somewhere during this to unlock the mouse. It will be nice if I can have a video of this."_

Ok, that's weird. What version of DisplayMagician does this happen with? 2.1.5.5, or the new one I just sent you 2.1.5.37? 

I've never had that issues you've described at all. DisplayMagician doesn't use Firefox, so I don't know how it could affect it particularly. What you're describing sounds like an issue with windows itself. I can't figure out why opening task manager could get things working again. Although now I think of it I've seen some behaviour like that from some malware as part of my day job. Do you have another computer to also install ATS and DisplayMagician on to see if the problem is on that computer too?  My gut feel is that there is an underlying problem with your windows installation, so installing ATS and DisplayMagician on another PC will help confirm that.  Maybe it is worth reinstalling windows on your PC if you don't have another one to test.

_"I don't know why DM is more stable if I start Steam on my profile but I think I should keep it because it's the only thing I found that resolve this bug. To prove my theory, I reactivated Steam on the tab 'choose waht happen before" and the bug is not present anymore after that. I don't understand this behavior after all that you said yesterday, it's very strange."_

You cannot use DisplayMagician to start steam as a start program. It is not designed to work that way, and will interfere with how DisplayMagician works. I have manually edited your Shortcuts_2.0.json and attached it here to give you the settings as they should be.
[Shortcuts_2.0.zip](https://github.com/terrymacdonald/DisplayMagician/files/7810149/Shortcuts_2.0.zip)

_"Anyway, to respond to your question, Profile B is nvidia surround (see attached screenshot for the names). You understand correctly, but I need to correct you in just one thing: After swap to my other profile display when I start the game (this second profile use nvidia surround), my primary display is still on my screen #1. I know this because if I start the game and switch back to my desktop when the game is still open, I see my notification area is still on screen #1, but I see the change during the roll back after I quit the game."_

OK. 

_"I tried what you said, I swaped my display profile from the displayprofile page and the problem is not present when I did this but the task bar disappear from screen #1 when I am on the surrund display (the task bar is only visible on screen #2 in this case)."_

OK - What I'm trying to do is figure out if it is a display profile issue, or an issue with the shortcut. You mentioned that the display profile swap problem isn't present when you use the Display Profile 'Apply' button to manually swap. You do get a problem with only one surround taskbar (I get that too on my Win 11 PC since I updated with the latest NVIDIA drivers), but otherwise it works properly.

If the above is true, then it means that the display profile part of DisplayMagician is working, as the game shortcut code actually uses the same display profile changing functions that the Apply button uses. 

_"After that, I tried to launch again the game with the DM sortcut and the behavior has changed. Now, if I use the shortcut, I see that DM have switched to my nvidia surround profile because I see the single complete taskbar in all the screen but the game appear only on my screen #1 and my screen #2 is on power save mode. The primary display is back to screen #1 after I quit the game (what I wanted) but the game don't use the 2 screen even if surround is activated."_

I need to see a DisplayMagician log for this situation in order to troubleshoot this. DisplayMagician wipes the previous log information after each restart, so can you please start DisplayMagician, then try and run the shortcut from the Shortcut Library window, then go to settings and create another support ZIP file. That should ensure that the right information is in the DisplayMagician log file and I should have more chance to figure out what is happening here.

_"I tried to relaunch the game with the DM shortcut and now, everythings works fine. The game start with the surround, and when I quit the game, my primary display is back to screen #1 (what I wanted). The problem seem to be not present after I swaped my dispaly profile from the displayprofile page."_

Wow, this makes no sense. It shouldn't fail to work properly and then work properly without a PC reboot. In the past when we've had display profile issues, they consistently didn't work. Yours is the strangest one yet.

_"After that, I did an utlimate test: I rebooted my computer, and use again the DM shortcut to start the game and the game have not started with the dual screen, only with the screen #1 even if I am on the nvidia surround profile (and screen #2 fall in power save mode again). The good thing is my screen #1 is still the primary display but the bad thing is (for the first time), the task bar on screen #1 have moved at the top of screen #1. If I look in the windows settings to the task bar, it siwtched to "custom" instead of "bottom" (see attached screenshot). So I suspect it's a bug with nvidia surround itself, what do you think?"_

I have had to taskbar jump to the top of the screen twice in Windows 11 since my last nvidia driver update. I also have had a windows update in that time, so it could be either of those. I found a reboot fixed it for me. 

_"Do you still need a second support zip file?"_

Heck yes. I really really need one.

I am still really confused with all the issues you're seeing. They seem like they could be windows related, nvidia driver related or something else. There isn't a pattern yet to how they happen, and the most important thing in troubleshooting bugs is to find the pattern, as that tells you where the bug is.

OK, I'd like you to do the following for me please to try and find a pattern to the bug:

- Please start DisplayMagician, then try and run the shortcut from the Shortcut Library window, then go to settings and create another support ZIP file. Upload that here.
- Close DisplayMagician again.
- Please restart DisplayMagician again then perform the following test:
  - Use the Display Profile window to apply the "1920x1080" display profile so that you are using that profile.
  - Create a new Game Shortcut to a different steam game (any game will do but DO NOT use ATS)
  - Set the Display Profile to the "3840x1080 Surround" display profile. 
  - Do not change the audio or microphone in the shortcut
  - Do not add any start programs
  - Set it so the display is reverted back after the game is closed(temporary change)
  - Save the new game shortcut
  - Run the new Shortcut from the Shortcut Library window (selected it an press the Run button)
  - go to settings and create another support ZIP file. Upload that 2nd support ZIP file here too.

That should then give me more information to troubleshoot. 

Thanks
Terry

**Comment by sk666** [comment]
Hi Terry,

just a little info for you, when I start nvidia surround by the nvidia control panel itself (not by DM), the taskbar disappear from screen #1 with the surround and it reappear when I disable it. So this little issue is related to nvidia and not to DM.

So before doing what you said in this issue, I began to uninstall DM, reboot, and did a clean install with the 2.1.5.37. Interesting fact, the 2.1.5.37 did not install over my previous version 2.1.3.1 (see attached screenshot), DM create another install who was the cause of some of the warnings messages on issue #75. So I uninstalled 2.1.5.37, tried to uninstall the 2.1.3.1 but failed, so I installed 2.1.3.1 again, rebbot, uninstall 2.1.3.1 (that worked), reboot, install 2.1.5.37 again (and I choose "do not close applications. A reboot will be required" during this install. The installation work like a charm without any warning, so rebooted agin and now, I followed your steps. Just to be sure, I erased my previous profiles ans shortcut to start with fresh one with your steps.

So, the first attemp was with ATS, and the second, like you said, was with a new shortcut with ETS 2. During the first attempt, I did not see DM applied surround so during the launch of ATS, ATS started on screen #1, screen $2 showed my desktop without entering in power save mode.

During my second attemp with ETS 2 shortcut, I saw DM applied surround, ETS 2 started on screen #1 and screen #2 switched to power save mode. 

Note: I did not use a desktop shortcut during this 2 test, I onlu used to run them in DM shortcut library window.

I hope that I did all of this correctly.

Thank you

![2 display magician installed](https://user-images.githubusercontent.com/29050341/148142550-d0229053-0f52-4f27-aea8-10b40d08d4b9.PNG)
[ATS DisplayMagician-Support-20220104-1922.zip](https://github.com/terrymacdonald/DisplayMagician/files/7811236/ATS.DisplayMagician-Support-20220104-1922.zip)
[ETS 2 DisplayMagician-Support-20220104-1925.zip](https://github.com/terrymacdonald/DisplayMagician/files/7811237/ETS.2.DisplayMagician-Support-20220104-1925.zip)
.

**Comment by terrymacdonald** [comment]
Hi @sk666,

Interesting. I hope we find some more info in the DM log files you've sent.

Can you please try with a game from a different game studio? So not an ETS2
or ATS game. Those games both use substantially similar code from the same
game studio, so I'd like to know how DM works with completely different
game from a different game studio. Can you please make another new shortcut
and do the same run shortcut, then create support zip file thing we've done
before?

Thanks
*Terry*

**Comment by sk666** [comment]
Hi Terry,

yeah sure. I did 2 test with 2 different game: Saint row 2 and Britannic: Patroness of the Mediterranean. both game needed a little update by steam so if you see that DM can not find the game, it is normal, it was because of these update. I did 2 test because saint row 2 seem to be to old for my computer ( it is the first time that I tried this game ever). With the britannic game, surround have srated correctly, game works well with the screen but when I roll back, the task bar on screen #1 have moved to the top of the screen just like you saw in W11. I will give you both logs just in case.

Thank you

**Comment by sk666** [comment]
[Sait Row 2 DisplayMagician-Support-20220104-2003.zip](https://github.com/terrymacdonald/DisplayMagician/files/7811367/Sait.Row.2.DisplayMagician-Support-20220104-2003.zip)
[Britannic DisplayMagician-Support-20220104-2006.zip](https://github.com/terrymacdonald/DisplayMagician/files/7811368/Britannic.DisplayMagician-Support-20220104-2006.zip)

**Comment by terrymacdonald** [comment]
Thanks @sk666.

Yeah both of those showed exactly what you said. The first game load was updating so DisplayMagician timed out (I'm sure I could find a fix for steam to detect updates somehow...something for later) but the subsequent game load worked fine. The display profiles applied properly, and there were no errors detected. To be honest I'm stumped now. 

I had a quick look in the NVIDIA forums to see if anyone had similar problems, and all I found was this: https://www.nvidia.com/en-us/geforce/forums/geforce-graphics-cards/5/480011/taskbar-moving/

Unfortunately there isn't really an easy for me to position the taskbar by brute force as part of a DisplayMagician config :(. I've looked into it before, and it was going to be really tough to get that to work reliably on Windows 10 and Windows 11.  https://www.howtogeek.com/760036/windows-11-lets-you-move-the-taskbar-to-the-left-or-right-but-its-broken/ has an example of doing this,  but handling this reliably over multiple monitors was just too tough to do :(.

OH! I just had an idea. NVIDIA may have changed some options in the latest driver update? Maybe you could try to create a new surround profile, and then swap to use that with the two ATS/ETS games? That might work? DisplayMagician doesn't create the NVIDIA configs, it just asks the NVIDIA Driver for a copy of the NVIDIA config, then saves it for later. This means that if NVIDIA change their config format, then it won't match our saved copy, and that causes problems. It could be (but is pretty unlikely) that NVIDIA changed something and that is causing an issue. Recreating the surround profile will guarantee that DisplayMagician has saved a new copy with this NVIDIA driver version.  Worth trying?

Hopefully the test version DisplayMagician v2.1.2.50 I linked to in issue #75 works better for you, and at least gets you most of the way to working? Please test it with all your changes and tell me if it works. [DisplayMagicianSetup-V2.1.5.50.zip](https://github.com/terrymacdonald/DisplayMagician/files/7820559/DisplayMagicianSetup-V2.1.5.50.zip)

Thanks
Terry

**Comment by sk666** [comment]
Hi Terry,

I will try with build 2.1.5.50. When you say "recreate the nvidia surround profile, you mean in the nvidia control pannel right? I alreday recreate it in DM but not in nvidia control pannel. Another thing I can test it's uninstall my nvidia driver, and do a clean install of the latest nvidia driver, just in case that I have some corruption in my current driver.

**Comment by sk666** [comment]
oh god I have big problem with the build .50. I tried to launch Britannic game to exclude all gfame from SCS software, DM activate the surround but never launched Steam neither the game itself. I'm stuck with the surround profile, I i try to select the display profil from de display profile page in DM, DM do nothing when I click on the apply button after I selected the 1920x1080 display profile. Nvidia control pannel see that surround is activated but I can not disable it by Nvidia control pannel, don't know why because I was able to do it 2 days ago. 

Also, I tried to change my display profile on the shortcut of Britannic game (in the game shortcut library as always to start it with the 1920x1080 display profile but that dosen't work, I'm stuck with the surround display with build .50 :(

**Comment by sk666** [comment]
I will give you the log just in case you need it.
[DisplayMagician-Support-20220106-1619.zip](https://github.com/terrymacdonald/DisplayMagician/files/7824820/DisplayMagician-Support-20220106-1619.zip)

**Comment by terrymacdonald** [comment]
I meant for you to:

   - Delete the 300x2000 surround profile you have (can't remember the name
   of it) in DisplayMagician
   - Turn off surround in NVIDIA Control Panel so it becomes individual
   screens again
   - Go back into NVIDIA Control Panel and set up the displays the way you
   want them
   - ALT-TAB back to DisplayMagician, then save the current display layout
   as a new Display Profile
   - Go in and edit all your surround Game Shortcuts to use the new Display
   Profile you just created.

Thanks

*Terry *

**Comment by terrymacdonald** [comment]
This is all beginning to sound like you have an underlying issue with
either NVIDIA Display Driver, or even with Windows itself. There are so
many strange things happening to your computer that neither myself or my
test users are experiencing (taskbar movement is the only one I think?)
that I really think it's time to try and see if an NVIDIA driver
uninstall/reinstall or a Windows reinstall fix the issue. I don't support
NVIDIA drivers or Windows, so I can't tell you to do it and I won't be able
to support you doing that, but if I were in the same situation I would
probably look at uninstalling the NVIDIA driver, restart and then reinstall
it. If that doesn't work, then I would look at reinstalling Windows 10
again. But that's just me.

If you do decide to do something similar to the above you will be on your
own. I can't help you with any guidance about reinstalling NVIDIA drivers
or reinstalling Windows. It needs to be your decision.

If you still are experiencing issues once you've tried fixing any
underlying issues please feel free to come back here and we'll restart some
troubleshooting then. If I don't hear from you for a few weeks then I'll
close the issue (though feel free to ask for it to be reopened if you need
it).

Thanks

*Terry MacDonald *| Principal Security Consultant | Chief Operating Officer



M: +64 211 918 814 <+64+211+918+814>
E: ***@***.***
W: www.cosive.com

**Comment by terrymacdonald** [comment]
Unfortunately this log just shows DisplayMagician starting up. The
DisplayMagician.log file is wiped and recreated each time DisplayMagician
runs. This means if DisplayMagician crashes, then we lose the log.

I will add an action item to see if there is a way I can keep the last 3
logs as that will help troubleshooting in the future.

Thanks

*Terry MacDonald *| Principal Security Consultant | Chief Operating Officer



M: +64 211 918 814 <+64+211+918+814>
E: ***@***.***
W: www.cosive.com

**Comment by sk666** [comment]
Ok so I did a uninstall of all nvidia related driver/software in safe mode, reboot, did a clean install to the latest nvidia driver. It seem a little bit more stable, at least I can disable my surround in nvidia control panel after activated it in DM build .50.  Previously, I should have restored my computer to a previous state with a windows system restore point to be able to do this, so, it's a nice improvement.

Build .50 seem to be problematic because DM works better with old version of DM, including the very old 2.0.1.59. With this old version, surround activate et deactivate when it should do it, also it start and exit steam by itself when it should do it so everything is fine with steam and surround with 2.0.1.59. 

But .50 and even .37 don't seem to work well with surround and they don't handle the roll back to my previous display settings, so what you want to do right know? Continue testing with my fresh nvidia install or if you are tired to debug this? I will undersatnd if you don't wanna go further with this, you gave me a lot of your time, I don't want to abuse it.

**Comment by sk666** [comment]
just a little update, build 50 seem to work with my 2 profile display in the display profile list in DM after my clean install of nvidia driver so, things are going in the right direction now :)

**Comment by terrymacdonald** [comment]
Ah that's good to hear! Progress at least.

Did you delete the old '3840x1080 Surround' display profile
from DisplayMagician and then recreate it through NVIDIA Control Panel? I'm
hoping that works.

I'm doubtful that there is an underlying problem with the NVIDIA
DisplayMagician library (NVIDIALibrary) when running a Surround configured
profile because:

   - I have multiple users running 2.1.5.50 using surround, and they run
   fine. There are also over 400 people running the release version of 2.1.5
   and no one else has complained that NVIDIA Surround is broken
   - In all my current testing with a GTX 3070 Ti all the features are
   successfully applied, other than the taskbar sometimes moving to the top of
   the primary screen. I have tested surround to surround profile swapping,
   with cloned displays, with non surround displays, with some screens on and
   with some screens off.
   - One of my users tested fine with 2.1.5 and had 7 screens connected,
   and even was doing some tricky things like having a surround screen with a
   couple of others turned on and some turned off.
   - When we tested earlier (please correct me if I have that wrong), your
   computer successfully swapped display profiles between '1920x1080' and
   '3840x1080 Surround' and back again fine when not part of a Game Shortcut.
   If this is still the case then it can't be an NVIDIA Surround issue you're
   experiencing.

There may still be an error in the application of the rollback profile, but
the current code looks like it should:

ApplyProfileResult result = ProfileRepository.ApplyProfile(rollbackProfile);

I think at this stage I should do some more digging in the
ShortcutRepository.RunShortcut
function to see if there is any possible place that DisplayMagician may be
getting the wrong display profile, or not storing it in a variable
correctly or anything like that. I'll get back to you with what I find.

Thanks

*Terry*

**Comment by terrymacdonald** [comment]
Oh I forgot to mention. One thing to do is to make sure that all your
Display Profiles have been recreated using 2.1.5.50. So if you recreate all
of them, then we know there isn't any residual settings left over from
previously. So, to be 100% sure, can you please perform the following
actions *in the specific order I've listed below*:

   - Load DisplayMagician
   - Delete all Display Profiles
   - Swap to NVIDIA Control Panel and turn off surround
   - Swap to Windows Display and set up the display layout for the
   '1920x1080' profile
   - Swap back to DisplayMagician and save the '1920x1080' Display Profile.
   - Swap back to NVIDIA Control Panel and turn on surround
   - Swap back to DisplayMagician and save the '3840x1080 Surround' Display
   Profile.
   - Use the 'Apply' button in the Edit Display Profile window to swap to
   the  '1920x1080' Display Profile.
   - Swap back to Windows Settings to verify that the Display Profile has
   successfully swapped back to the '1920x1080' display layout
   - Swap back to DisplayMagician and use the 'Apply' button in the Edit
   Display Profile window to swap to the '3840x1080 Surround' Display Profile.
   - Swap back to NVIDIA Control Panel and check that the surround layout
   is exactly as you expect it.
   -
   - Use the 'Apply' button in the Edit Display Profile window to swap back
   to the  '1920x1080' Display Profile.
   - Swap back to Windows Settings to verify that the Display Profile has
   successfully swapped back to the '1920x1080' display layout

Once that is sorted and tested, then the next step is to update your Game
Shortcuts to make use of the new DisplayProfiles. They will all be showing
error symbols on them because they point to a non-existing display profile.

   - Go to the Shortcut Library window
   - Edit each Game Shortcut in turn, and pick the new  '3840x1080
   Surround' Display Profile *for all of them (just for testing)*
   - Save the Game Shortcut
   - Repeat for all the other Game Shortcuts until they have all been
   updated.

At this point, we want to try and run the shortcuts and see what happens.

   - Make sure you are currently in the '1920x1080' Display Profile. We
   need to start from a layout we know is 'correct'.
   - Run the 'Britannic: Patroness of the Mediterranean' game.
   - Check that DisplayMagician changes to the  '3840x1080
   Surround' profile, and runs the game correctly.
   - Wait until the game starts, and then leave it 30 seconds or so.
   - Exit the Britannic game
   - DisplayMagician should detect the shutdown, and should revert back to
   the  '1920x1080' Display Profile.
   - Go back to the Edit Display Profile window to check that
   DisplayMagician thinks it's back on the '1920x1080' Display Profile. (If
   you click on the 'View Current' button it should select the '1920x1080'
   Display Profile from the list)

If all that happens, then we know that the display profile swapping is
working ok!

Please follow the above instructions exactly, as each step relies on the
previous step being done to make sure the test is accurate. *If you do get
an error, go straight to the Settings page and create a Support ZIP file so
I get the logs.*

Hope that all helps!

Thanks

*Terry *

**Comment by sk666** [comment]
Hi Terry,

just before I uninstalled my nvidia driver yesterday, I completely uninstalled DM including the folder in the appdata/local to test an old version like I said yesterday. After that, I completely uninstalled again DM including the appdata/local folder, unistalled nvidia driver and reinstalled them, and then, install DM .50. When I started it, Dm was completly empty,m no shortcut (the button was missing because I didn't have any display profile created and obsviousily, the display profile windows in Dm was empty like it should be.

After reading this, do you need these test or it will be the same? I can do it if you want, it's not a problem for me, I'm asking just to be sure.

Another point: I tried to break my settings with my fresh install of nvidia driver and DM with .50 and hopefully, I dont have any issue except that steam won't launch by itself with ATS shortcut in DM shortcut window. Another bug (just for your information because it is not caused by DM), 1 task bar seem to disappear from one of the screen in surrund, it happens if I switch to surrund in nvidia control panel and not just with DM so it's obvioulsy a glitch with nvidia, I am not the only one with this bug, and personnaly, I can live with that. I don,t use surround except when I play games so I don't need my taskbar during this time. The taskbar never moved again after my fresh install of nvidia driver, so may be it's the solution for this bug... 

So at time i'm writing this, I don't have any glitch with issue #74 but you can add the steam dosen't launch by itself in the issue #75. Things are way better since yesterday. So I suppose that issue #74 was due to some corruption in the nvidia driver.

Thank you

**Comment by terrymacdonald** [comment]
Hi @sk666,

Yes, I agree it does seem like there is a glitch in the NVIDIA driver. We'll have to keep an eye on it and hope it fixes itself over the next few driver releases.

I'll close down issue #74 for now, but as I mentioned before, feel free to reopen it if you find the same problem happening again. If you get a different problem please log a new separate issue so we can track that separately.

Thanks
Terry
